### PR TITLE
refactor: parallelize invariants

### PR DESCRIPTION
**Pull request by DaniPopes**
## Motivation

Most of the invariant code is written as if it handles multiple functions per call (e.g. argument is `&[Function]` instead of `&Function` etc.).

This is fine, except we call these functions with `vec![function]` in a loop over the functions...

Recommended to review with "Hide whitespace" since this gets rid of a lot of indentation

## Solution

Rewrite invariants to run one function at a time, and parallelize the top level loop

### CODE REVIEW

File: crates/forge/src/runner.rs [hunk 1]
```diff
@@ -407,129 +409,134 @@ impl<'a> ContractRunner<'a> {
...
         );
 
         let invariant_contract =
-            InvariantContract { address, invariant_functions: functions, abi: self.contract };
+            InvariantContract { address, invariant_functions: func, abi: self.contract };
 
-        let Ok(InvariantFuzzTestResult { invariants, cases, reverts, last_run_inputs }) =
-            evm.invariant_fuzz(&invariant_contract)
-        else {
-            return vec![]
```
Comment (DaniPopes) [inline crates/forge/src/runner.rs:457]: This error was just completely ignored for some reason. This gets thrown by `prepare_fuzzing`

**Review by Evalir — APPROVED** [review]
I love this—there's a ton we can refactor here but this is great to reduce performance bottlenecks 😄

Let's just make sure we test this a ton before merge!

**Review by Evalir — APPROVED** [review]
I love this—there's a ton we can refactor here but this is great to reduce performance bottlenecks 😄

Let's just make sure we test this a ton before merge!

File: crates/forge/src/runner.rs (hunk 1 above)
Comment (Evalir) [reply crates/forge/src/runner.rs:457]: Yeah, I think there's a few more cases of this happening. I have a pending refactor of another error being ignored on invariants

**Review by mattsse — APPROVED** [review]
amazing

File: crates/evm/src/fuzz/invariant/executor.rs [hunk 2]
```diff
@@ -622,48 +606,31 @@ fn can_continue(
 
             failures.revert_reason = Some(error.revert_reason.clone());
 
-            // Hacky to provide the full error to the user.
-            for invariant in invariant_contract.invariant_functions.iter() {
-                failures.failed_invariants.insert(
-                    invariant.name.clone(),
-                    (Some(error.clone()), invariant.to_owned().clone()),
-                );
-            }
```
Comment (Rubilmax) [inline crates/evm/src/fuzz/invariant/executor.rs:631]: Why was this removed?

Comment (Rubilmax) [reply crates/evm/src/fuzz/invariant/executor.rs:631]: Can you take a look at #5726 pls? I think a bug was introduced but I can't manage to find when or where
